include/xtensor/xpad.hpp

```diff
@@ -81,82 +81,104 @@ namespace xt
+                xt::xstrided_slice_vector svs(e.shape().size(), xt::all());
+                xt::xstrided_slice_vector svt(e.shape().size(), xt::all());
+
+                svt[axis] = xt::range(0, nb);
+
+                if (mode == pad_mode::wrap || mode == pad_mode::periodic)
+                {
+                    XTENSOR_ASSERT(nb <= e.shape(axis));
+                    svs[axis] = xt::range(e.shape(axis), nb+e.shape(axis));
+                }
+                else if (mode == pad_mode::symmetric)
+                {
+                    XTENSOR_ASSERT(nb <= e.shape(axis));
+                    svs[axis] = xt::range(2*nb-1, nb-1, -1);
+                }
+                else if (mode == pad_mode::reflect)
+                {
+                    XTENSOR_ASSERT(nb <= e.shape(axis) - 1);
+                    svs[axis] = xt::range(2*nb, nb, -1);
+                }
+
+                xt::strided_view(out, svt) = xt::strided_view(out, svs);
             }
-            else
+
+            if (ne > static_cast<size_type>(0))
             {
-                xt::xstrided_slice_vector sv_bgn(e.shape().size(), xt::all());
-                xt::xstrided_slice_vector sv_end(e.shape().size(), xt::all());
+                xt::xstrided_slice_vector svs(e.shape().size(), xt::all());
+                xt::xstrided_slice_vector svt(e.shape().size(), xt::all());
+
+                svt[axis] = xt::range(out.shape(axis)-ne, out.shape(axis));
 
                 if (mode == pad_mode::wrap || mode == pad_mode::periodic)
                 {
-                    XTENSOR_ASSERT(nb <= out.shape()[axis]);
-                    XTENSOR_ASSERT(ne <= out.shape()[axis]);
-                    sv_bgn[axis] = xt::range(out.shape()[axis]-nb, out.shape()[axis]);
-                    sv_end[axis] = xt::range(0, ne);
+                    XTENSOR_ASSERT(ne <= e.shape(axis));
+                    svs[axis] = xt::range(nb, nb+ne);
                 }
                 else if (mode == pad_mode::symmetric)
                 {
-                    XTENSOR_ASSERT(nb <= out.shape()[axis]);
-                    XTENSOR_ASSERT(ne <= out.shape()[axis]);
-                    if (nb == size_type(0))
-                    {
-                        sv_bgn[axis] = xt::range(0, 0, -1);
-                    }
-                    else
-                    {
-                        sv_bgn[axis] = xt::range(nb-1, _, -1);
-                    }
-                    if (ne == out.shape()[axis])
+                    XTENSOR_ASSERT(ne <= e.shape(axis));
+                    if (ne == nb + e.shape(axis))
                     {
-                        sv_end[axis] = xt::range(out.shape()[axis], _, -1);
+                        svs[axis] = xt::range(nb+e.shape(axis)-1, _, -1);
                     }
                     else
                     {
-                        sv_end[axis] = xt::range(out.shape()[axis], out.shape()[axis]-ne-1, -1);
```

Comment:
@JohanMabille What surprised me a bit is that in hindsight I think that originally this might not have correct (although the code passed the relevant tests). It would be good if you can double check.

Reply:
Indeed, I think the range in the first version should have started at `out.shape()[axis] - 1`. This is weird. However I am sure the change is correct.